src/arcaflow_plugin_sdk/atp.py

```diff
@@ -143,40 +144,43 @@ def run_server_read_loop(self) -> None:
             while True:
                 # Decode the message
                 runtime_msg = self.decoder.decode()
-                msg_id = runtime_msg["id"]
-                run_id = runtime_msg["run_id"]
+                msg_id = runtime_msg.get("id", None)
                 # Validate
                 if msg_id is None:
-                    self.stderr.write("Runtime message is missing the 'id' field.")
+                    self.send_error_message("", step_fatal=False, server_fatal=True,
+                                            error_msg="Runtime message is missing the 'id' field.")
```

Comment:
Personally I think the indenting continuation on the `(` is ugly, wastes space, and makes the whole thing harder to read; but in this case it also seems inconsistent with the wrapping of other argument lists in the module. I'm kinda surprised that your `black` is accepting both.